t/29-backend-svirt.t

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/perl
+
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+use Test::Most;
+use Mojo::Base -strict, -signatures;
+use Test::Warnings qw(:all :report_warnings);
+use Test::Fatal;
+use Test::MockModule;
+use Test::MockObject;
+use Test::Mock::Time;
+use Test::Output qw(stderr_like);
+use Mojo::File qw(tempdir);
+use Mojo::Util qw(scope_guard);
+use FindBin '$Bin';
+use lib "$Bin/../external/os-autoinst-common/lib";
+use OpenQA::Test::TimeLimit '5';
+use backend::svirt;
+use distribution;
+
+my $dir = tempdir("/tmp/$FindBin::Script-XXXX");
+chdir $dir;
+my $cleanup = scope_guard sub { chdir $Bin; undef $dir };
+
+my $ssh_object = Test::MockObject->new();
+$ssh_object->set_true(qw/disconnect scp_get blocking/);
+$ssh_object->set_always('error', 'Mock SSH Error');
+my $chan_object = Test::MockObject->new();
+$chan_object->set_true(qw/exec/);
+
+
+
+my $run_ssh_cmd_mock = Test::MockModule->new('backend::baseclass');
+$run_ssh_cmd_mock->redefine(new_ssh_connection => sub ($self, %args) { return $ssh_object });
+$run_ssh_cmd_mock->redefine(start_ssh_serial => sub { return ($ssh_object, $chan_object) });
+
+sub redefine_ssh ($excepted = 0) {
+    $run_ssh_cmd_mock->redefine(run_ssh_cmd => sub ($self, $cmd, %args) {
+            return ref($excepted) eq 'ARRAY' ? @$excepted : $excepted;
+    });
+}
+
+$bmwqemu::vars{WORKER_HOSTNAME} = 'localhost';
+$bmwqemu::vars{VIRSH_HOSTNAME} = 'foobar';
+$bmwqemu::vars{VIRSH_USERNAME} = 'root';
+$bmwqemu::vars{VIRSH_PASSWORD} = 'password';
+$bmwqemu::vars{JOBTOKEN} = "JobToken";
+
+my $distri = $testapi::distri = distribution->new();
+
+subtest 'Generic svirt backend' => sub {
+    my $backend = backend::svirt->new;
+    my $bmwqemu_mock = Test::MockModule->new('bmwqemu');
+    # silence some log output for cleaner tests
+    $bmwqemu_mock->noop('diag');
+    $bmwqemu_mock->noop('log_call');
+    redefine_ssh;
+    $backend->{need_delete_log} = 1;
+    ok $backend->do_start_vm, 'can start vm';
+    is $backend->can_handle({function => 'snapshots'}), undef, 'can handle snapshots';
```

Comment:
The expected value is `undef` and but the test is called 'can handle snapshots'. Either `undef` should be `{ret => 1}` or 'can handle snapshots' should be 'can not handle snapshots' for this to make sense.

The same counts for the subsequent tests. They assume a falsy value is correct but the test name sounds positive.

Reply:
yes , this is exactly what worries me ...

culprit:
`if ($args->{function} eq 's...' && !bmwqmenu::vars{HDDFORMAT} eq 'raw') {}`

for any value of `HDDFORMAT` it returns **FALSE**

when I modified condition to use `bmwqmenu::vars{HDDFORMAT} ne 'raw'` it works as intended but when is `HDDFORMAT` undefined throws exception ...

so I tried find where and how is used this variable -> only in `backend::qemu` to kill backend as unsupported var

Reply:
`!bmwqmenu::vars{HDDFORMAT} eq 'raw'` looks wrong, indeed. It negates `bmwqmenu::vars{HDDFORMAT}` and *then* does the equality comparison. I'd also switch it to `ne`. To handle the `undef` case, I'd use `(bmwqmenu::vars{HDDFORMAT} // '') ne 'raw'`.

Reply:
I'm more in favor to remove `HDDFORMAT`. And also from documentation from backend cars